front/src/views/user/eating/eating-new/eating-new.view.js

```diff
@@ -0,0 +1,171 @@
+import { MmtTextInput } from '@/components';
+import { requiredValidationMixin } from '@mixins';
+import { LINK_REGEX, MAX_FILE_SIZE_SCHEDULET_PLACE_TO_EAT } from '@constants';
+import {
+  VCol,
+  VRow,
+  VBtn,
+  VIcon,
+  VTextField,
+  VMenu,
+  VDatePicker,
+  VTimePicker,
+  VForm,
+  VFileInput
+} from 'vuetify/lib';
+
+import { gmapApi } from 'vue2-google-maps';
+export default {
+  mixins: [ requiredValidationMixin ],
+  components: {
+    VCol,
+    VRow,
+    VBtn,
+    VIcon,
+    VTextField,
+    VMenu,
+    VDatePicker,
+    MmtTextInput,
+    VTimePicker,
+    VForm,
+    VFileInput
+  },
+  data () {
+    return {
+      menuDatePicker: false,
+      menuTimePicker: false,
+      formValid: false,
+      rules: {
+        maxLengthName: v => v.length <= 200 || this.$t('eatingNew.maxLengthName'),
+        maxLengthNotes: v => v.length <= 2000 || this.$t('eatingNew.maxLengthNotes'),
+        linkValid: v => {
+          if (v) {
+            return RegExp(LINK_REGEX).test(v) || this.$t('eatingNew.isNoLink');
+          } else {
+            return true;
+          }
+        }, // правило на 10 мб
+        fileValid: v => {
+          if (v) {
+            let valid = true;
+            v.forEach(file => {
+              if (file.size > MAX_FILE_SIZE_SCHEDULET_PLACE_TO_EAT) {
+                valid = false;
+              }
+            });
+            return valid ? true : 'File size should be less than 10 MB!';
+          } else {
+            return true;
+          }
```

Comment:
could it be useful for other components? if yes - it is nice to have mixin

Reply:
ok